File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.cc
#include <iostream>
#include <string>
#include <vector>
using namespace std;

bool isGroupWord(const string& word) {
    bool visited[26] = { false };
    
    char prev = 0;
    for (char c : word) {
        if (c != prev) {
            if (visited[c - 'a']) {
                return false;
            }
            visited[c - 'a'] = true;
        }
        prev = c;
    }
    
    return true;
}

int main() {
    int N;
    cin >> N;
    
    int count = 0;
    while (N--) {
        string word;
        cin >> word;
        
        if (isGroupWord(word)) {
            count++;
        }
    }
    
    cout << count << '\n';
    return 0;
}

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.swift
import Foundation

func isGroupWord(_ word: String) -> Bool {
    var visited = Array(repeating: false, count: 26)
    var prev: Character = " "

    for ch in word {
        if ch != prev {
            let index = Int(ch.asciiValue! - Character("a").asciiValue!)
            if visited[index] {
                return false
            }
            visited[index] = true
        }
        prev = ch
    }

    return true
}

let n = Int(readLine()!)!
var count = 0

for _ in 0..<n {
    let word = readLine()!
    if isGroupWord(word) {
        count += 1
    }
}

print(count)